commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java

```diff
@@ -81,6 +83,24 @@ public int prefixLength() {
 		return prefixLength;
 	}
 	
+	public Boolean synonym() {
+		return synonym;
+	}
+	
+	public TextPredicate withSynonyms(Boolean enableSynonyms) {
+		this.synonym = enableSynonyms;
```

Comment:
I'd suggest using the same name consistently (with-er method suffix, local parameter, field name), as occasionally we resort to method calls using reflection and a systematic naming convention helps a lot:

```suggestion
	public TextPredicate withSynonym(Boolean synonym) {
		this.synonym = synonym;
```

Reply:
How about keeping the plural form here @apeteri and only changing the method argument name?

Reply:
The accessor and the field is also singular: https://github.com/b2ihealthcare/snow-owl/blob/48d0fa99fbed809fa96930f2d412436e1cc4e180/commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java#L86-L88

I'm not married to either form, whatever makes sense to you. 😺

Reply:
`synonymsEnabled` ?

Reply:
Sounds good!